### DISCUSSION TITLE
Is there any way to set a custom NameProvider or other way to change the default Cross Reference behavior?

### DISCUSSION THREAD

**Discussion by AignerGames** [eclipse-langium/langium#1734] [Q&A] [answered]
Hello,
currently the property "name" is used to find cross references, but I would like to use a different name. 
I think a way to change this would be to provide a custom NameProvider implementation?
But Im new to langium and have the default template and no idea where I would inject a custom implementation into the whole system.

**Comment by msujew** [comment, marked as answer]
Hey @AignerGames,

Yes you're completely right, you would create a new `NameProvider` (or override the existing `DefaultNameProvider` and use that in your service module. See [our documentation](https://langium.org/docs/reference/configuration-services/#overriding-and-extending-services) on how to do that.

**Reply by AignerGames** [reply]
Thank you for the fast answer. I somehow missed that part of the documentation. I checked the example and managed to add my custom NameProvider, so this question is solved.
